tests/storage/test_unsafe_locale.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from unittest.mock import patch
+
+from synapse.storage.database import make_conn
+from synapse.storage.engines._base import IncorrectDatabaseSetup
+from synapse.storage.engines.postgres import PostgresEngine
+
+from tests.unittest import HomeserverTestCase
+
+
+class UnsafeLocaleTest(HomeserverTestCase):
```

Comment:
You will probably need to add the import for this, but looking at other tests with `rg` (since I vaguely remembered seeing it), this is how we usually skip Postgres-specific tests.
That way, they're neither marked as a failure nor a success on SQLite, which I guess is pedantically more correct.
Would imagine it avoids the overhead of setting up a test homeserver + database just to not do anything :-).

```suggestion
class UnsafeLocaleTest(HomeserverTestCase):
    if not USE_POSTGRES_FOR_TESTS:
        skip = "Requires Postgres"

```